distributed/utils_test.py

```diff
@@ -879,6 +894,7 @@ async def end_worker(w):
     await asyncio.gather(*(end_worker(w) for w in workers))
     await s.close()  # wait until scheduler stops completely
     s.stop()
+    check_invalid_worker_transitions(s)
```

Comment:
I understand that transition failures are catastrophic events, but this makes me wonder whether there are other events that should be checked for during testing fixture teardown. Why not test for them here too? Should we also be serialising them in `Scheduler.events` as a matter of policy and reconstituting them here? Is this a once-off special-case handling strategy for transition failures?

Reply:
Possibly, but for now I'd like to focus on these events.  If we find that we want to do this a couple more times then we'll probably generalize things.  For now I'd like to focus on a concrete problem for which we have an immediate need of a solution.  If we try to do something general then we'll be guessing I think about what the right structure is.

Reply:
I think this is fine as exploratory code to tease out failure. I'm concerned that it becomes a forgotten special case. Do we have a process for keeping track of this kind of thing (another github issue for instance?)